Gems/Atom/RHI/Code/Source/RHI/FrameGraphCompiler.cpp

```diff
@@ -111,14 +111,6 @@ namespace AZ::RHI
 
         FrameGraph& frameGraph = *request.m_frameGraph;
 
-        for (Scope* scope : frameGraph.GetScopes())
```

Comment:
It has been moved to `FrameScheduler::PrepareProducers()` to happen a bit earlier since we need it for proper Scope(Attachment) linkage and we save on the additional loop over the scopes.